tests/internal/test_monitoring_tool.py

```diff
@@ -0,0 +1,224 @@
+
+    request.addfinalizer(finalizer)
+    clone_repo(
+        constants.ODF_MONITORING_TOOL_REPO, str(repo_dir), branch="main", tmp_repo=True
+    )
+    return repo_dir
+
+
+def convert_yaml_file_to_json_file(file_path):
+    """
+    Util to convert yaml file to json file and replace an ext of the files
+
+    Args:
+        file_path (Path): path to the file to convert
+    Returns:
+        Path: path to the new file, yaml converted to json
+    """
+    logger.info(f"convert yaml file '{file_path}' to json format")
+    with open(file_path) as file:
+        content = yaml.safe_load(file)
+    new_path = file_path.parent / (file_path.with_suffix("").name + ".json")
+    os.rename(file_path, new_path)
+    with open(new_path, "w") as file:
+        json.dump(content, file)
+    return new_path
+
+
+def comparetool_deviation_check(first_file, second_file, deviation_list):
+    """
+    Function to run 'comparetool' and verify deviation_list accordingly to comparetool output
+
+    Args:
+        first_file (Path): path to the file, standing first in comparetool args to compare
+        second_file (Path): path to the file, standing second in comparetool args to compare
+        deviation_list (list): list of deviation strings expected to be in comparetool output
+    """
+    complete_proc = exec_cmd(f"./comparealerts {first_file} {second_file}")
+    compare_res = complete_proc.stdout.decode("utf-8")
+    assert all(
+        alert in compare_res for alert in deviation_list
+    ), f"compare tool did not find all occurancies from {deviation_list}:\n{compare_res}"
+
+
+@tier1
+@blue_squad
+@polarion_id("OCS-4844")
+class TestMonitoringTool(BaseTest):
+    def test_odf_monitoring_tool(
+        self,
+        clone_odf_monitoring_compare_tool,
+        clone_ocs_operator,
+        clone_upstream_ceph,
+    ):
+
+        check_go_version()
+
+        os.chdir(clone_odf_monitoring_compare_tool / "comparealerts")
+        exec_cmd("go build")
+
+        prometheus_ocs_rules = (
+            clone_ocs_operator / "metrics" / "deploy" / "prometheus-ocs-rules.yaml"
+        )
+        prometheus_alerts_upstream = (
+            clone_upstream_ceph / "monitoring" / "ceph-mixin" / "prometheus_alerts.yml"
+        )
+
+        assert os.path.isfile(
+            prometheus_ocs_rules
+        ), f"cannot find '{prometheus_ocs_rules}' in cloned repo"
+        assert os.path.isfile(
+            prometheus_alerts_upstream
+        ), f"cannot find '{prometheus_alerts_upstream}' in cloned repo"
+        logger.debug(f"print '{prometheus_ocs_rules}' file")
+        exec_cmd(f"cat {prometheus_ocs_rules}")
+        logger.debug(f"print '{prometheus_alerts_upstream}' file")
+        exec_cmd(f"cat {prometheus_alerts_upstream}")
+
+        logger.info("compare upstream and downstream prometheus rule files")
+        complete_proc = exec_cmd(
+            f"./comparealerts {prometheus_ocs_rules} {prometheus_alerts_upstream}"
```

Comment:
https://github.com/red-hat-storage/ocs-operator.git is not a downstream repo. This test doesn't take into account version of installed cluster, only what is in master of the repo.

Reply:
this test doesn't require cluster, and any version of the OCP, ODF deployment accordingly. 
relatively one to another and between two files:
prometheus_ocs_rules - came from our 'downstream' repo https://github.com/red-hat-storage/ocs-operator.git
prometheus_alerts_upstream - came from the 'upstream' ceph repo https://github.com/ceph/ceph.git

that is what I understood from conversation with devel assignee